x-pack/plugins/beats_management/public/lib/adapters/framework/kibana_framework_adapter.ts

```diff
@@ -0,0 +1,184 @@
+  public get info() {
+    if (this.xpackInfo) {
+      return this.xpackInfo;
+    } else {
+      throw new Error('framework adapter must have init called before anything else');
+    }
+  }
+
+  public get currentUser() {
+    return this.shieldUser!;
+  }
+  private xpackInfo: FrameworkInfo | null = null;
+  private adapterService: KibanaAdapterServiceProvider;
+  private shieldUser: FrameworkUser | null = null;
+  constructor(
+    private readonly PLUGIN_ID: string,
+    private readonly management: ManagementSetup,
+    private readonly getBasePath: () => string,
+    private readonly licensing: LicensingPluginSetup,
+    private readonly securitySetup: SecurityPluginSetup | undefined,
+    private readonly config: BeatsManagementConfigType,
+    public readonly version: string
+  ) {
+    this.adapterService = new KibanaAdapterServiceProvider();
+  }
+
+  public setUISettings = (key: string, value: any) => {
+    this.adapterService.callOrBuffer(({ config }) => {
+      config.set(key, value);
+    });
+  };
+
+  public async waitUntilFrameworkReady(): Promise<void> {
+    const license = await this.licensing.license$.pipe(first()).toPromise();
+    let xpackInfoUnpacked: FrameworkInfo;
+
+    try {
+      xpackInfoUnpacked = {
+        basePath: this.getBasePath(),
+        license: {
+          type: license.type ?? 'oss',
+          expired: !license.isActive,
+          expiry_date_in_millis: license.expiryDateInMillis ?? -1,
+        },
+        security: {
+          enabled: license.getFeature('security').isEnabled,
+          available: license.getFeature('security').isAvailable,
+        },
+        settings: this.config,
+      };
+    } catch (e) {
+      throw new Error(`Unexpected data structure from xpackInfoService, ${JSON.stringify(e)}`);
+    }
+
+    const assertData = RuntimeFrameworkInfo.decode(xpackInfoUnpacked);
+    if (isLeft(assertData)) {
+      throw new Error(
+        `Error parsing xpack info in ${this.PLUGIN_ID},   ${PathReporter.report(assertData)[0]}`
+      );
+    }
+    this.xpackInfo = xpackInfoUnpacked;
+
+    try {
+      this.shieldUser = (await this.securitySetup?.authc.getCurrentUser()) || null;
+      const assertUser = RuntimeFrameworkUser.decode(this.shieldUser);
+
+      if (isLeft(assertUser)) {
+        throw new Error(
+          `Error parsing user info in ${this.PLUGIN_ID},   ${PathReporter.report(assertUser)[0]}`
+        );
+      }
+    } catch (e) {
+      this.shieldUser = null;
+    }
+  }
+
+  public registerManagementSection(settings: {
+    id: string;
+    name: string;
+    iconName: string;
+    order?: number;
+  }) {
+    this.management.sections.register({
+      id: settings.id,
+      title: settings.name,
+      euiIconType: settings.iconName,
+      order: settings.order || 30,
+    });
+  }
+
+  public registerManagementUI(settings: {
+    sectionId: string;
+    appId: string;
+    name: string;
+    basePath: string;
+    visable?: boolean;
```

Comment:
basePath & visable are unused

Reply:
Yeah I can delete these I guess, was just doing minimal changes. Also love the spelling errors here 😆